File: watermark-api/src/main/java/com/watermark/watermarkapi/domains/WatermarkUrl.java
package com.watermark.watermarkapi.domains;

public class WatermarkUrl {

    private final String algorithm;
    private final Integer imageId;

    public WatermarkUrl(Integer imageId, String algorithm) {
        this.imageId = imageId;
        this.algorithm = algorithm;
    }

    public Integer getImageId() {
        return imageId;
    }

    public String getAlgorithm() {
        return algorithm;
    }
}

File: watermark-client/src/app/domains/WatermarkUrl.ts
export class WatermarkUrl {
  url: string;
  algorithm: string;

  constructor(url: string, algorithm: string) {
    this.url = url;
    this.algorithm = algorithm;
  }

  public getUrl(): string {
    return this.url;
  }

  public getAlgorithm(): string {
    return this.algorithm;
  }
}
